### ISSUE TITLE
Incorrect link for inherited java method from collection

### ISSUE THREAD

**Issue by vmishenev** [Kotlin/dokka#2879]
- In stdlib AbstractMutableCollection: the functions like `spliterator, stream, toArray, toString` point anywhere except that `AbstractMutableCollection` member pages: some point to `AbstractMutableSet`, some - to `LinkedHashSet`.

- Another reproducer ([the unit test](https://github.com/Kotlin/dokka/blob/d31fc588dcba45bd0cbbc413419fdf4a3911cb39/plugins/base/src/test/kotlin/model/InheritorsTest.kt#L450)):
```kt
            
            import java.util.AbstractCollection

            abstract class MyCollection<E> : AbstractCollection<E>()
```

**Comment by vmishenev** [comment]
I have opened [KT-56838](https://youtrack.jetbrains.com/issue/KT-56838/JavaMethodDescriptor.overriden.overriden-has-an-incorrect-value-for-inherited-methods-of-Java-collections)

**Comment by IgnatBeresnev** [comment]
Might be related: #2464

**Comment by IgnatBeresnev** [comment]
The problem is on our side in the place where internal links (anchor) are resolved.

Typealises have their own pages now, so the internal link resolution might not be needed anymore, and this issue might be fixed by removing this resolution.

The anchor link resolution is somewhat confusing, so it might take longer to fix.

**Comment by vmishenev** [comment]
Related issue: #3054

**Comment by vmishenev** [comment]
#3054 can fix it, but in this case it would cause #3542.

**Comment by IgnatBeresnev** [comment]
Try to filter out such J2K declarations. If there's no API to do that (esp. in Analysis API), create an upstream issue for it, and change the status of this issue to blocked. Descriptors (K1) - optional

**Comment by vmishenev** [comment]
This is fixed by #3694 as well
